packages/core/src/components/forms/_control-group.scss

```diff
@@ -189,6 +189,12 @@ Styleguide control-group
     border-radius: 0 $pt-border-radius $pt-border-radius 0;
   }
 
+  // round all the corners of the only child element
+  > :only-child {
```

Comment:
well color me 😱 i did not know this pseudoselector existed!
and it's always been there. https://developer.mozilla.org/en-US/docs/Web/CSS/:only-child